interpreter/src/main/scala/pl/touk/nussknacker/engine/Interpreter.scala

```diff
@@ -262,15 +258,19 @@ object Interpreter {
 
     override def monad: Monad[IO] = Monad[IO]
 
-    override def fromFuture[T](implicit ec: ExecutionContext): Future[T] => IO[Either[T, Throwable]] =
-      f => IO.fromFuture(IO.pure(transform(f)))(IO.contextShift(ec))
+    override def fromFuture[T]: Future[T] => IO[Either[T, Throwable]] = {
+      implicit val ctx = SynchronousExecutionContext.ctx
+      f => IO.fromFuture(IO(transform(f)))(IO.contextShift(ctx))
+    }
   }
 
   class FutureShape(implicit ec: ExecutionContext) extends InterpreterShape[Future] {
 
     override def monad: Monad[Future] = cats.instances.future.catsStdInstancesForFuture(ec)
 
-    override def fromFuture[T](implicit ec: ExecutionContext): Future[T] => Future[Either[T, Throwable]] = transform(_)
+    override def fromFuture[T]: Future[T] => Future[Either[T, Throwable]] = {
+      transform(_)(SynchronousExecutionContext.ctx)
```

Comment:
Let's be consistent and do `implicit val ctx = SynchronousExecutionContext.ctx`?

Reply:
nope, there is already `ec` implicit in scope

Reply:
So, we we pass this `SynchronousExecutionContext.ctx`?

Reply:
as you can see, in the previous implementation, two execution contexts were passed:
1. in the constructor - it was used to create a monad instance
2. in the `fromFuture` method - it was used to do the `transform`. 

In this change, I don't want to remove the constructor execution context, because it's not related to any of the issues I found.